pkg/controllers/user/logging/clusterlogging.go

```diff
@@ -76,59 +124,77 @@ func (c *ClusterLoggingSyncer) Sync(key string, obj *v3.ClusterLogging) error {
 		if err != nil {
 			return err
 		}
+
 		if !isAllDisable {
 			utils.UnsetConfigMap(c.configmaps, loggingconfig.ClusterLoggingName, "cluster")
 		}
 
-		var updateErr error
 		if obj != nil && !reflect.DeepEqual(obj.Spec, obj.Status.AppliedSpec) {
-			updatedObj := obj.DeepCopy()
-			updatedObj.Status.AppliedSpec = obj.Spec
-			_, updateErr = c.clusterLoggings.Update(updatedObj)
+			return unsetClusterLogging(obj, c.clusterLoggings)
 		}
-		return updateErr
+		return nil
 	}
 
-	original := obj
-	obj = original.DeepCopy()
-
-	newObj, err := c.doSync(obj)
-	if err != nil {
-		return err
+	if ok, delay := c.backoffFailure(obj); ok {
```

Comment:
the backoff stuff is majorly used to forget this kind of error and avoid consistent failures in the sync(). I feel like the mechanism is complicated. What if we don't take the `waitingmsg` case as an error and get rid of the mechanism. We don't set the conditon to False in that case now(`https://github.com/aiwantaozi/rancher/blob/62e807392d26130a68c673aa201d3f167f7a9355/pkg/controllers/user/logging/clusterlogging.go#L307`)